lib/spack/spack/build_systems/python.py

```diff
@@ -346,16 +349,20 @@ def headers(self) -> HeaderList:
         # Remove py- prefix in package name
         name = self.spec.name[3:]
 
-        # Headers may be in either location
+        # Headers should only be in include or platlib, but no harm in checking purelib too
         include = self.prefix.join(self.spec["python"].package.include).join(name)
         platlib = self.prefix.join(self.spec["python"].package.platlib).join(name)
-        headers = fs.find_all_headers(include) + fs.find_all_headers(platlib)
+        purelib = self.prefix.join(self.spec["python"].package.purelib).join(name)
+
+        find_all_headers = functools.partial(fs.find_all_headers, recursive=True)
```

Comment:
@wdconinc There is an issue here. `find_all_headers` has a different signature than `find all_*_libraries` :sweat: and takes no `recursive=...` argument.

Reply:
Let me submit a quick fix